pkg/sdk/streams_def.go

```diff
@@ -1,9 +1,54 @@
 package sdk
 
-import g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+import (
+	"fmt"
+	"strings"
+
+	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+)
 
 //go:generate go run ./poc/main.go
 
+type StreamSourceType string
+
+const (
+	StreamSourceTypeTable         StreamSourceType = "TABLE"
+	StreamSourceTypeExternalTable StreamSourceType = "EXTERNAL TABLE"
+	StreamSourceTypeView          StreamSourceType = "VIEW"
+	StreamSourceTypeStage         StreamSourceType = "STAGE"
+)
+
+func ToStreamSourceType(s string) (StreamSourceType, error) {
```

Comment:
nit: we usually test these funcs.

Reply:
I don't think we usually do, but I added tests here.

Reply:
We do, if you know about some places when we don't, then we should also add them there.